src/providers/AppLinkProvider.tsx

```diff
@@ -211,11 +212,26 @@ const useAppLink = () => {
             scanResult = {
               type,
               payees: Object.entries(rawScanResult.payees).map((entries) => {
-                const scanData = entries[1] as { amount: string; memo?: string }
+                let amount
+                let memo
+                if (entries[1]) {
+                  if (typeof entries[1] === 'number') {
+                    // Case (4) stringified JSON object { type, payees: {[payeeAddress]: amount} }
+                    amount = entries[1] as number
+                  } else if (typeof entries[1] === 'object') {
+                    // Case (5) stringified JSON object { type, payees: {[payeeAddress]: { amount, memo? }} }
+                    const scanData = entries[1] as {
+                      amount: string
+                      memo?: string
+                    }
+                    amount = scanData.amount
+                    memo = scanData.memo
+                  }
+                }
                 return {
                   address: entries[0],
-                  amount: scanData.amount,
-                  memo: scanData.memo,
+                  amount: `${amount}`,
```

Comment:
If `amount` was actually defined as a number in the QR code JSON data, then this later triggers a crash due to a JS runtime error (in production you see a blank, dark-blue screen, in development you see the screen below)
![image](https://user-images.githubusercontent.com/3493682/122607888-e269c900-d040-11eb-9ec3-7ce8d959b8c3.png)

Reply:
Nice good catch, thanks for fixing this